### DISCUSSION TITLE
React Native Support

### DISCUSSION THREAD

**Discussion by MuckT** [WalletConnect/walletconnect-monorepo#2928] [Developer Support] [answered]
### Description

I am attempting to get the `chat-client` working with React Native; however, I am getting errors with the crypto library not being present on react native.

```
 {"context": "core"} [TypeError: undefined is not an object (evaluating 'crypto.web.subtle.digest')]
 {"context": "chatClient"} undefined is not an object (evaluating 'crypto.web.subtle.digest')
```

**Comment by crypblizz8** [comment]
@devceline @Cali93 👀

**Comment by Cali93** [comment, marked as answer]
Hey @MuckT,

We are working on a React Native SDK for Web3Inbox which is built on top of chat-client and is more convenient for developers.
That being said, to get `chat-client` working on React Native, you'll need to polyfill `crypto`.

1. Install and set up [react-native-quick-crypto](https://www.npmjs.com/package/react-native-quick-crypto) to polyfill crypto or `expo-crypto` if you're using [expo](https://docs.expo.dev/versions/latest/sdk/crypto/).

2. Add the following snippet in the `index.js` file of your react-native app
```js
import crypto from 'react-native-quick-crypto';

const polyfillDigest = async (algorithm, data) => {
  const algo = algorithm.replace('-', '').toLowerCase();
  const hash = crypto.createHash(algo);
  hash.update(data);
  return hash.digest();
};

globalThis.crypto = crypto;
globalThis.crypto.subtle = {
  digest: polyfillDigest,
};
```
